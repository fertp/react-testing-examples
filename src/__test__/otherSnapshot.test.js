import renderer from 'react-test-renderer'
import { render, screen, getByText } from '@testing-library/react'

const Header = () => {
  return (
    <header data-testid='header'>
      <h1>Soy un Títulos</h1>
    </header>
  )
}

test('Snapshot with testing-library', () => {
  const { container } = render(<Header/>)
  expect(container).toMatchSnapshot()
})

test('Snashot with react-test-renderer', () => {
  const tree = renderer.create(<p>Hola</p>).toJSON()
  expect(tree).toMatchInlineSnapshot(`
  <p>
    Hola
  </p>
  `)
})



test('Seleccionar elementos anidados', () => {
  // render(<Header/>)
  // const header = screen.getByTestId('header')
  // const title = getByText(header, 'Soy un Título')
  // expect(title).toBeInTheDocument()
})