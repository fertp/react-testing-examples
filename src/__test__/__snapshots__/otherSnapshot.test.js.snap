// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot with testing-library 1`] = `
<div>
  <header
    data-testid="header"
  >
    <h1>
      Soy un TÃ­tulos
    </h1>
  </header>
</div>
`;
